### DISCUSSION TITLE
How to clone prisma client object?

### DISCUSSION THREAD

**Discussion by rajesh1158** [prisma/prisma#14525] [Q&A]
How can we clone the prisma client object so that we can re-use the same client across all worker threads (worker_threads module) in nodejs application? Node js requires an object be cloneable if we need to share it among worker threads?

**Comment by nurul3101** [comment]
Hi @rajesh1158 👋 

To reuse your `PrismaClient` object and the generated model types you could create a private npm package. This package would export `PrismaClient` object and the types, I have seen this approach being used by many users. Would this work for you?

In case this doesn't work, could you try using the [cloneDeep](https://lodash.com/docs/4.17.15#cloneDeep) method of lodash to create a clone of PrismaClient?

**Comment by rajesh1158** [comment]
What I meant by reusing the prisma client is the instance of the client `new PrismaClient()`. So, I cannot use a private npm package.

The lodash deep clone returned an empty object, so this also does not work.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
